app/coffee/DrainManager.coffee

```diff
@@ -0,0 +1,25 @@
+logger = require "logger-sharelatex"
+
+module.exports =
+	startDrain: (io, rate) ->
+		# Clear out any old interval
+		clearInterval @interval
+		if rate == 0
+			return
+		@interval = setInterval () =>
+			@reconnectNClients(io, rate)
+		, 1000
+
+	RECONNECTED_CLIENTS: {}
+	reconnectNClients: (io, N) ->
+		drainedCount = 0
+		for client in io.sockets.clients()
+			if !@RECONNECTED_CLIENTS[client.id]
+				@RECONNECTED_CLIENTS[client.id] = true
+				logger.log {client_id: client.id}, "Asking client to reconnect gracefully"
+				client.emit "reconnectGracefully"
+				drainedCount++
+			if drainedCount == N
+				break
```

Comment:
a touch unclear about this break. Maybe me, i just get confused what will happen. 

could do something like this on line 17

maxClientsDrained = drainedCount == N
if !maxClientsDrained and !@RECONNECTED_CLIENTS[client.id]